### PR TITLE
Added support to use Pathlib

### PR DESCRIPTION
**Pull request by SigireddyBalasai**
for the issue #90 added the pathlib support

### CODE REVIEW

**Comment by SigireddyBalasai** [comment]
@microsoft-github-policy-service agree

File: src/markitdown/_markitdown.py
Comment (l-lumin) [inline src/markitdown/_markitdown.py:16]: what is the purpose of  `overload`?

File: src/markitdown/_markitdown.py
Comment (l-lumin) [inline src/markitdown/_markitdown.py:1046]: could you update the docstring as well?

File: tests/test_markitdown.py
Comment (l-lumin) [inline tests/test_markitdown.py:324]: the tests seem repetitive. Could you wait until the test refactor PR is approved and merged and add tests for pathlib later?

**Review by l-lumin — CHANGES_REQUESTED** [review]

File: src/markitdown/_markitdown.py
Comment (SigireddyBalasai) [reply src/markitdown/_markitdown.py:16]: i forget to remove it after trying to use different approach

File: tests/test_markitdown.py
Comment (SigireddyBalasai) [reply tests/test_markitdown.py:324]: i need to wait for sometime before adding the tests right

File: src/markitdown/_markitdown.py
Comment (SigireddyBalasai) [reply src/markitdown/_markitdown.py:1046]: done

**Comment by SigireddyBalasai** [comment]
@l-lumin please check the new changes

File: tests/test_markitdown.py
Comment (afourney) [inline tests/test_markitdown.py:10]: Are we using Path elsewhere in this tests file?

Otherwise, PR looks good.

**Review by l-lumin — APPROVED** [review]

File: tests/test_markitdown.py
Comment (SigireddyBalasai) [reply tests/test_markitdown.py]: no usage of it

Comment (SigireddyBalasai) [reply tests/test_markitdown.py]: removed it thank you

**Review by SigireddyBalasai — COMMENTED** [review]
done the change

**Comment by SigireddyBalasai** [comment]
@afourney please let me know of status

**Review by gagb — APPROVED** [review]

File: src/markitdown/_markitdown.py
Comment (gagb) [inline src/markitdown/_markitdown.py:1311]: Is there a test for this?

**Review by gagb — APPROVED** [review]

**Pull request merged**
